log4mongo/test/test_handlers.py

```diff
@@ -196,4 +197,170 @@ def test_emit_on_reconnect_existing_capped_collection(self):
+    collection_name = 'buffered_logs_test'
+
+    def setUp(self):
+        self.handler = BufferedMongoHandler(host=self.host_name,
+                                    database_name=self.database_name,
+                                    collection=self.collection_name,
+                                    buffer_size=5, buffer_periodical_flush_timing=None,
+                                    buffer_early_flush_level=logging.CRITICAL
+                                    )
+        self.log = logging.getLogger('testLogger')
+        self.log.setLevel(logging.DEBUG)
+        self.log.addHandler(self.handler)
+        self.old_stderr = sys.stdout
+        sys.stderr = StringIO()
+
+    def force_flush(self):
+        self.handler.flush_to_mongo()
+
+    def test_contextual_info(self):
+        self.log.info('test message with contextual info',
+                      extra={'ip': '127.0.0.1', 'host': 'localhost'})
+        self.force_flush()
+        document = self.handler.collection.find_one(
+            {'message': 'test message with contextual info', 'level': 'INFO'})
+        self.assertEqual(document['message'],
+                         'test message with contextual info')
+        self.assertEqual(document['level'], 'INFO')
+        self.assertEqual(document['ip'], '127.0.0.1')
+        self.assertEqual(document['host'], 'localhost')
+
+    def test_contextual_info_adapter(self):
+        adapter = logging.LoggerAdapter(self.log,
+                                        {'ip': '127.0.0.1',
+                                         'host': 'localhost'})
+        adapter.info('test message with contextual info')
+        self.force_flush()
+        document = self.handler.collection.find_one(
+            {'message': 'test message with contextual info', 'level': 'INFO'})
+        self.assertEqual(document['message'], 'test message with contextual info')
+        self.assertEqual(document['level'], 'INFO')
+        self.assertEqual(document['ip'], '127.0.0.1')
+        self.assertEqual(document['host'], 'localhost')
+
+    def test_emit(self):
+        self.log.warning('test message')
+        self.force_flush()
+        document = self.handler.collection.find_one(
+            {'message': 'test message', 'level': 'WARNING'})
+        self.assertEqual(document['message'], 'test message')
+        self.assertEqual(document['level'], 'WARNING')
+
+    def test_emit_exception(self):
+        try:
+            raise Exception('exc1')
+        except:
+            self.log.exception('test message')
+        self.force_flush()
+
+        document = self.handler.collection.find_one(
+            {'message': 'test message', 'level': 'ERROR'})
+        self.assertEqual(document['message'], 'test message')
+        self.assertEqual(document['level'], 'ERROR')
+        self.assertEqual(document['exception']['message'], 'exc1')
+
+    def test_emit_fail(self):
+        self.handler.collection = ''
+        self.log.warn('test warning')
```

Comment:
Ah true, I copied and pasted code of test_emit_fail from same method of parent class. I will update both if you want.

Also I have see some tests that are basically not doing the right thing, I will try to correct it as well.